code/controllers/subsystem/processing/orbits.dm

```diff
@@ -117,30 +118,39 @@ PROCESSING_SUBSYSTEM_DEF(orbits)
 		//Update UIs
 		for(var/datum/tgui/tgui as() in open_orbital_maps)
 			tgui.send_update()
+	update_ruins()
+	update_objectives()
+	update_events()
+	//Do processing.
+	if(!resumed)
+		. = ..()
+		if(MC_TICK_CHECK)
+			return
+		//Update UIs
+		for(var/datum/tgui/tgui as() in open_orbital_maps)
+			tgui.send_update()
```

Comment:
limits what I can add after it in the future, however most things should go before UI updates.
```suggestion
	if(resumed)
		return
	. = ..()
	if(MC_TICK_CHECK)
		return
	//Update UIs
	for(var/datum/tgui/tgui as() in open_orbital_maps)
		tgui.send_update()
```